Framework/PythonInterface/test/python/mantid/geometry/ComponentInfoTest.py

```diff
@@ -105,5 +106,112 @@ def test_name(self):
         info = self._ws.componentInfo()
         self.assertEquals(type(info.name(0)), str)
 
+
+    """ 
+    The section below is for destructive tests - i.e extreme
+    and exceptional test cases. 
+    """
+
+    def test_detectorsInSubtree_exceptional(self):
+        info = self._ws.componentInfo()
+        with self.assertRaises(TypeError):
+        	info.detectorsInSubtree("Error")
+        with self.assertRaises(TypeError):
+        	info.detectorsInSubtree(10.0)
+        with self.assertRaises(Exception):
+        	info.detectorsInSubtree(100)
+
+    def test_componentsInSubtree_exceptional(self):
+	    info = self._ws.componentInfo()
+	    with self.assertRaises(TypeError):
+	    	info.componentsInSubtree("Error")
+	    with self.assertRaises(TypeError):
+	    	info.componentsInSubtree(10.0)
+
+    def test_size_exceptional(self):
+    	info = self._ws.componentInfo()
+    	with self.assertRaises(TypeError):
+    		info.size(0)
+
+    def test_isDetector_exceptional(self):
+    	info = self._ws.componentInfo()
+    	with self.assertRaises(TypeError):
+    		info.isDetector("Error")
+    	with self.assertRaises(TypeError):
+    		info.isDetector(10.0)
+    	self.assertEquals(info.isDetector(10000), False)
+
+    def test_hasDetectorInfo_exceptional(self):
+        info = self._ws.componentInfo()
+        with self.assertRaises(TypeError):
+        	info.hasDetectorInfo(0)
+
+    def test_position_exceptional(self):
+		info = self._ws.componentInfo()
+		with self.assertRaises(TypeError):
+			info.position("Zero")
+		with self.assertRaises(TypeError):
+			info.position(0.0)
+
+    def test_rotation_exceptional(self):
+		info = self._ws.componentInfo()
+		with self.assertRaises(TypeError):
+			info.rotation("Zero")
+		with self.assertRaises(TypeError):
+			info.rotation(0.0)
```

Comment:
It would be more valuable for many of these to check that the accepted ranges of indices are allowed. i.e. what if I ask for index 100 when there are only 100 components? Same with index -1.

Reply:
Oh okay. I can add those in now.